crates/vm2-interface/src/state_interface.rs

```diff
@@ -1,93 +1,148 @@
     fn set_transaction_number(&mut self, value: u16);
 
+    /// Returns the value of the context register.
     fn context_u128_register(&self) -> u128;
+    /// Sets the value of the context register.
     fn set_context_u128_register(&mut self, value: u128);
 
+    /// Iterates over storage slots read or written during VM execution.
     fn get_storage_state(&self) -> impl Iterator<Item = ((H160, U256), U256)>;
-
+    /// Iterates over all transient storage slots set during VM execution.
     fn get_transient_storage_state(&self) -> impl Iterator<Item = ((H160, U256), U256)>;
+    /// Gets value of the specified transient storage slot.
     fn get_transient_storage(&self, address: H160, slot: U256) -> U256;
+    /// Sets value of the specified transient storage slot.
     fn write_transient_storage(&mut self, address: H160, slot: U256, value: U256);
 
+    /// Iterates over events emitted during VM execution.
     fn events(&self) -> impl Iterator<Item = Event>;
+    /// Iterates over L2-to-L1 logs emitted during VM execution.
     fn l2_to_l1_logs(&self) -> impl Iterator<Item = L2ToL1Log>;
 
+    /// Gets the current amount of published pubdata.
     fn pubdata(&self) -> i32;
+    /// Sets the current amount of published pubdata.
     fn set_pubdata(&mut self, value: i32);
 }
 
+/// VM execution flags. See the EraVM reference for more details.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub struct Flags {
+    /// "Less than" flag.
     pub less_than: bool,
+    /// "Equal" flag.
     pub equal: bool,
+    /// "Greater than" flag.
     pub greater: bool,
 }
 
+/// Public interface of an EraVM call frame.
 pub trait CallframeInterface {
+    /// Address of the storage context associated with this frame. For delegate calls, this address is inherited from the calling contract;
+    /// otherwise, it's the same as [`Self::code_address()`].
     fn address(&self) -> H160;
+    /// Sets the address of the executing contract.
     fn set_address(&mut self, address: H160);
+    /// Address of the contract being executed.
     fn code_address(&self) -> H160;
+    /// Sets the address of the contract being executed.
     fn set_code_address(&mut self, address: H160);
+    /// Address of the calling contract. Respects delegate and mimic calls.
     fn caller(&self) -> H160;
+    /// Sets the address of the calling contract
     fn set_caller(&mut self, address: H160);
 
-    /// During panic and arbitrary code execution this returns None.
+    /// Returns the current program counter (i.e., 0-based index of the instruction being executed).
+    /// During panic and arbitrary code execution this returns `None`.
     fn program_counter(&self) -> Option<u16>;
-
+    /// Sets the program counter.
     /// The VM will execute an invalid instruction if you jump out of the program.
     fn set_program_counter(&mut self, value: u16);
 
+    /// Returns a 0-based index of the starting exception handler instruction.
```

Comment:
This and the comment for set use very different terminology. How about something like "Returns the program counter that the parent frame continue from if this frame fails"